session.go

```diff
@@ -1660,58 +1662,75 @@ func (c *Collection) EnsureIndexKey(key ...string) error {
 //     http://www.mongodb.org/display/DOCS/Multikeys
 //
 func (c *Collection) EnsureIndex(index Index) error {
-	if index.Sparse && index.PartialFilter != nil {
-		return errors.New("cannot mix sparse and partial indexes")
-	}
+	return c.EnsureIndexes([]Index{index})
+}
 
-	keyInfo, err := parseIndexKey(index.Key)
-	if err != nil {
-		return err
-	}
+// EnsureIndexes ensures all indexes exist submitting one batch command with any new and/or changed indexes to Mongo for best performance.
+//
+// See EnsureIndex for details on the index specifications.
+func (c *Collection) EnsureIndexes(indexes []Index) error {
+	indexSpecs := []indexSpec{}
 
 	session := c.Database.Session
-	cacheKey := c.FullName + "\x00" + keyInfo.name
-	if session.cluster().HasCachedIndex(cacheKey) {
-		return nil
-	}
+	for _, index := range indexes {
+		if index.Sparse && index.PartialFilter != nil {
+			return errors.New("cannot mix sparse and partial indexes")
+		}
 
-	spec := indexSpec{
-		Name:                    keyInfo.name,
-		NS:                      c.FullName,
-		Key:                     keyInfo.key,
-		Unique:                  index.Unique,
-		Background:              index.Background,
-		Sparse:                  index.Sparse,
-		Bits:                    index.Bits,
-		Min:                     index.Minf,
-		Max:                     index.Maxf,
-		BucketSize:              index.BucketSize,
-		ExpireAfter:             int(index.ExpireAfter / time.Second),
-		Weights:                 keyInfo.weights,
-		DefaultLanguage:         index.DefaultLanguage,
-		LanguageOverride:        index.LanguageOverride,
-		Collation:               index.Collation,
-		PartialFilterExpression: index.PartialFilter,
-	}
+		keyInfo, err := parseIndexKey(index.Key)
+		if err != nil {
+			return err
+		}
 
-	if spec.Min == 0 && spec.Max == 0 {
-		spec.Min = float64(index.Min)
-		spec.Max = float64(index.Max)
-	}
+		cacheKey := c.FullName + "\x00" + keyInfo.name
+		if session.cluster().HasCachedIndex(cacheKey) {
+			continue
+		}
 
-	if index.Name != "" {
-		spec.Name = index.Name
-	}
+		spec := indexSpec{
+			Name:                    keyInfo.name,
+			NS:                      c.FullName,
+			Key:                     keyInfo.key,
+			Unique:                  index.Unique,
+			Background:              index.Background,
+			Sparse:                  index.Sparse,
+			Bits:                    index.Bits,
+			Min:                     index.Minf,
+			Max:                     index.Maxf,
+			BucketSize:              index.BucketSize,
+			ExpireAfter:             int(index.ExpireAfter / time.Second),
+			Weights:                 keyInfo.weights,
+			DefaultLanguage:         index.DefaultLanguage,
+			LanguageOverride:        index.LanguageOverride,
+			Collation:               index.Collation,
+			PartialFilterExpression: index.PartialFilter,
+			logicalKeyName:          keyInfo.name,
```

Comment:
```logicalKeyName``` is the same as ```Name```, so you can use that instead. 
Alternatively, you could store the cache key.